launchpad/launchpad.gno

```diff
@@ -0,0 +1,377 @@
+package launchpad
+
+import (
+	"errors"
+	"std"
+	"strconv"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/ufmt"
+	u256 "gno.land/p/gnoswap/uint256"
+	"gno.land/r/gnoswap/v1/common"
+	"gno.land/r/gnoswap/v1/consts"
+	en "gno.land/r/gnoswap/v1/emission"
+	"gno.land/r/gnoswap/v1/gns"
+)
+
+var (
+	projects = make(map[string]Project) // projectId -> project
+
+	// project tier should distrubte project token if deposit ever happend
+	// therefore we need to keep track of project tiers without deposit
+	projectTiersWithoutDeposit = make(map[string]bool) // tierId -> true
+)
+
+var (
+	q96 = u256.Zero()
```

Comment:
how about init actual value here

Reply:
because of the syntax error, this was unavoidable.